### PR TITLE
Fix memory leak in actor building block

### PR DESCRIPTION
**Pull request by youngbupark**
# Description

With profiling daprd, we found the below issues:
1. http client in app healthz check consumes unexpected memory because it creates client whenever it sends request.
2. app healthz check timeout is too short to get the response back when app is busy state. healthy check is flaky. This led to disconnecting placement frequently, forcing daprd rebalance actors frequently.

Proposed fix:
1. use single http client connection pool
2. make a room for health check period and request timeout 
3. do not reconnect placement when app is unhealthy.

> Note: For test, since we will refactor and change placement client logic for Raft change, I will postpoen to write the test for the changes (we do not have test for placement connection test even today)

## Issue reference

#2093

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/actors/actors.go [hunk 1]
```diff
@@ -151,7 +151,15 @@ func (a *actorsRuntime) Init() error {
 	log.Infof("actor runtime started. actor idle timeout: %s. actor scan interval: %s",
 		a.config.ActorIdleTimeout.String(), a.config.ActorDeactivationScanInterval.String())
 
-	go a.startAppHealthCheck()
+	// Be careful to configure healthz endpoint option. If app healthz returns unhealthy status, Dapr will
+	// disconnect from placement to remove the node from consistent hashing ring.
+	// i.e if app is busy state, the heathz status would be flaky, which leads to frequent
+	// actor reblanacing. It will impact the entire service.
```
Comment (artursouza) [inline pkg/actors/actors.go:157]: typo.: rebalancing

Comment (youngbupark) [reply pkg/actors/actors.go]: fixed.

**Review by artursouza — APPROVED** [review]

**Comment by youngbupark** [comment]
image

This PR reduce the memory usage significantly. Once turning Metrics and traces off, memory usage is reduced as well.

**Review by artursouza — APPROVED** [review]

**Pull request merged**